clash-prelude/src/Clash/Magic.hs

```diff
@@ -0,0 +1,38 @@
+{-|
+  Copyright   :  (C) 2019, Myrtle Software Ltd
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <clash@qbaylogic.com>
+
+Control module instance, and register, names in generated HDL code.
+-}
+
+{-# LANGUAGE DataKinds      #-}
+{-# LANGUAGE ExplicitForAll #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE TypeOperators  #-}
+
+module Clash.Magic where
```

Comment:
`Clash.NameMagic`?

Reply:
No, I want to add more magic here eventually